drivers/gpio/gpio_gecko.c

```diff
@@ -19,6 +19,22 @@
 	CONFIG_GPIO_INIT_PRIORITY.
 #endif
 
+#if DT_NODE_HAS_PROP(id, peripheral_id)
+#define GET_GECKO_GPIO_INDEX(id) DT_INST_PROP(id, peripheral_id)
+#else
+#if defined(CONFIG_SOC_SERIES_EFR32BG22) || defined(CONFIG_SOC_SERIES_EFR32MG21)
+#define GECKO_GPIO_PORT_ADDR_SPACE_SIZE sizeof(GPIO_PORT_TypeDef)
+#else
+#define GECKO_GPIO_PORT_ADDR_SPACE_SIZE sizeof(GPIO_P_TypeDef)
+#endif /* defined(CONFIG_SOC_SERIES_EFM32HG) || defined(CONFIG_SOC_SERIES_EFM32WG) */
+/* Assumption:
+ * 1. Address space of the first GPIO port starts at GPIO peripheral
+ *    base address which is taken from GPIO_BASE HAL macro
+ */
+#define GET_GECKO_GPIO_INDEX(id) (DT_INST_REG_ADDR(id) - GPIO_BASE) \
```

Comment:
`GPIO_BASE` -> `DT_REG_ADDR(DT_NODELABEL(gpioa))`, no HAL needed for addresses.

Reply:
Changed `GPIO_BASE` to `DT_REG_ADDR(DT_NODELABEL(gpioa))`